python_modules/libraries/dagster-looker/dagster_looker/asset_decorator.py

```diff
@@ -0,0 +1,31 @@
+import itertools
+from pathlib import Path
+from typing import Any, Callable
+
+import yaml
+from dagster import AssetKey, AssetsDefinition, AssetSpec, multi_asset
+
+
+def looker_assets(*, project_dir: Path) -> Callable[[Callable[..., Any]], AssetsDefinition]:
+    lookml_dashboard_specs = [
+        AssetSpec(
+            key=AssetKey(lookml_dashboard["dashboard"]),
+            deps={AssetKey(dashboard_element["explore"])},
+        )
+        for dashboard_path in project_dir.rglob("*.dashboard.lookml")
+        # Each dashboard file can contain multiple dashboards.
+        for lookml_dashboard in yaml.safe_load(dashboard_path.read_bytes())
```

Comment:
at first glance the triple-comprehension was a little hard to follow, especially with the filter condition, might be easier to parse split into a few intermediate steps (& would probably be more self-documenting/reduce need for inline comments)